monai/networks/utils.py

```diff
@@ -688,27 +690,26 @@ def convert_to_onnx(
             onnx_inputs = (inputs,)
         else:
             onnx_inputs = tuple(inputs)
-
         if filename is None:
-            f = io.BytesIO()
+            temp_file = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
Hi @borisfom, based on your comments here: https://github.com/Project-MONAI/MONAI/issues/8149#issuecomment-2433177783
I try to create a temp file when filename is None, could you please take a look at whether the change here make senses? Thanks.

Reply:
Well the way it is now, it would crash if filename in not None on temp_file.close() below 
I suggest not using delete=False then close/remove is not needed. You may do temp_file=None above to keep it in the scope : 
```
temp_file = None
if filename is None:
            temp_file = tempfile.NamedTemporaryFile(delete=False)
            ...
```

Reply:
Updated, please help take a look at it again, thanks!